### ISSUE TITLE
(Web) Sections with parenthesis removed from `href` in <Link />

### ISSUE THREAD

**Issue by kimchouard** [expo/expo#26852]
### Minimal reproducible example

https://github.com/kimchouard/link-href-error-web/blob/main/components/EditScreenInfo.tsx

### Which package manager are you using? (Yarn is recommended)

yarn

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

None

### Summary

If a link has some parenthesis in its path, the section containing them will be removed. 
➡️ E.g. `http://testurl.com/path/with(parenthesis)/file.ext` will link to `http://testurl.com/path/file.ext` when used in a `<Link href={...} />`. 

_Note: this behavior is only on the Web, works fine on native._

I know it's a byproduct of the file-base routing system and its groups (e.g. `app/(admin)/index.tsx`), but now that `<Link />` can be used for external links with `target="_blank"`, this automation could lead to unexpected behavior on the Web. Even in the `tabs@50` template, the `<ExternalLink />` demo component (based on `<Link />` too) removes the `(...)` sections. 😬

👉 See this [demo](https://expo-link-web-bug.netlify.app/) ([source code](https://github.com/kimchouard/link-href-error-web/blob/main/components/EditScreenInfo.tsx)) built straight from the `tabs@50` template.

---

Proposed behavior:
- When `href` is an absolute path (starting with `/`), keep current behavior.
- When `href` is an "external" path (i.e. starts with `http://`, `https://`, `://`...), do not remove sections with `(...)` parenthesis (since it won't be an Expo Router path).

I tried to look for where this href "sanitation" was happening to create a PR but couldn't find it. 🤷🏼‍♂️ I searched around here:
https://github.com/expo/expo/blob/957904e9987bf5590826954584e4b45330d42086/packages/expo-router/src/link/path.ts#L31

### Environment

Doesn't really depend on the env, but here's mine:

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 13.6.3
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.3.0 - /opt/homebrew/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 9.6.7 - /opt/homebrew/bin/npm
      Watchman: 2023.09.04.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.2 - /Users/kimchouard/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
    IDEs:
      Android Studio: 2022.1 AI-221.6008.13.2211.9619390
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.5 => 50.0.5 
      expo-router: ~3.4.6 => 3.4.6 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.2 => 0.73.2 
      react-native-web: ~0.19.6 => 0.19.10 
    npmGlobalPackages:
      eas-cli: 7.0.0
      expo-cli: 6.3.8
    Expo Workflow: managed

**Comment by kimchouard** [comment]
Interestingly, it works in Expo Snacks, see [this one](https://snack.expo.dev/@kimik/dfd0d5) I just created. But definitely fails locally or when pushed online (via Netfly or so).

**Comment by kimchouard** [comment]
**TEMP FIX:**  encode your URLs by replacing `(` by `%28` and `)` by `%29`.

```js
const expoLinkProofingExternalUrls = (url) => url?.replace(/\(/g, '%28').replace(/\)/g, '%29');
```

This will keep the `href` value untouched via `<Link />`

**Comment by marklawlor** [comment]
This is resolved in `3.5.0`, which is part of the upcoming SDK51

**Comment by kimchouard** [comment]
Great, thanks @marklawlor ! 🙌🏻
